Sources/Datadog/FeaturesIntegration/WebView/WebRUMEventConsumer.swift

```diff
@@ -7,7 +7,81 @@
 import Foundation
 
 internal class WebRUMEventConsumer: WebEventConsumer {
-    func consume(event: [String: Any], eventType: String) {
-        // TODO: RUMM-1791 implement event consumers
+    private let dataWriter: Writer
+    private let dateCorrector: DateCorrectorType
+    private let contextProvider: RUMContextProvider?
+
+    init(
+        dataWriter: Writer,
+        dateCorrector: DateCorrectorType,
+        contextProvider: RUMContextProvider?
+    ) {
+        self.dataWriter = dataWriter
+        self.dateCorrector = dateCorrector
+        self.contextProvider = contextProvider
+    }
+
+    func consume(event: JSON, eventType: String) throws {
+        let eventData = try JSONSerialization.data(withJSONObject: event, options: [])
+        let jsonDecoder = JSONDecoder()
+        let rumContext = contextProvider?.context
+
+        switch eventType {
+        case "view":
+            let viewEvent = try jsonDecoder.decode(RUMViewEvent.self, from: eventData)
+            let mappedViewEvent = mapIfNeeded(dataModel: viewEvent, context: rumContext, offset: getOffset(viewID: viewEvent.view.id))
+            write(mappedViewEvent)
+        case "action":
+            let actionEvent = try jsonDecoder.decode(RUMActionEvent.self, from: eventData)
+            let mappedActionEvent = mapIfNeeded(dataModel: actionEvent, context: rumContext, offset: getOffset(viewID: actionEvent.view.id))
+            write(mappedActionEvent)
+        case "resource":
+            let resourceEvent = try jsonDecoder.decode(RUMResourceEvent.self, from: eventData)
+            let mappedResourceEvent = mapIfNeeded(dataModel: resourceEvent, context: rumContext, offset: getOffset(viewID: resourceEvent.view.id))
+            write(mappedResourceEvent)
+        case "error":
+            let errorEvent = try jsonDecoder.decode(RUMErrorEvent.self, from: eventData)
+            let mappedErrorEvent = mapIfNeeded(dataModel: errorEvent, context: rumContext, offset: getOffset(viewID: errorEvent.view.id))
+            write(mappedErrorEvent)
+        case "long_task":
+            let longTaskEvent = try jsonDecoder.decode(RUMLongTaskEvent.self, from: eventData)
+            let mappedLongTaskEvent = mapIfNeeded(dataModel: longTaskEvent, context: rumContext, offset: getOffset(viewID: longTaskEvent.view.id))
+            write(mappedLongTaskEvent)
+        default:
+            userLogger.error("🔥 Web RUM Event Error - Unknown event type: \(eventType)")
+        }
```

Comment:
What if we add a new RUM event type in Browser SDK version `X` (loaded by the webview), but mobile app is using old version of iOS SDK with this outdated `switch eventType` selection? It seems that we will be dropping all new events. // cc @mariusc83 @xgouchet do we mitigate this on Android?

Reply:
Maybe instead of decoding events to concrete types (`RUMViewEvent`, `RUMActionEvent`, ...) we should only decode their mutable parts to single type, e.g.:
```swift
struct RUMBrowserEvent: Codable {
   // MARK: - Mutable, known properties
   var applicationID: String
   var sessionID: String

   // MARK: - Immutable, unknown properties
   let otherProperties: [String: Codable]
}
```
This way we can overwrite only `sessionID` and `applicationID` without changing anything else to the event:
```swift
/*

Say, `eventData` is:
{
   "application": { "id": "browser-application-id" },
   "session": { "id": "browser-session-id" },
   "other_stuff" {
      "foo": 100,
      "bar": "bizz",
   },
   "more_stuff": "fuzz"
}

*/

// Then mutating it:
var browserRUMEvent = try jsonDecoder.decode(RUMBrowserEvent.self, from: eventData)
browserRUMEvent.applicationID = rumContext.applicationID
browserRUMEvent.sessionID = rumContext.sessionID

// ... and serializing it again:
write(browserRUMEvent)

/*

... will alter only 'application.id' and 'session.id':
{
   "application": { "id": "native-application-id" },
   "session": { "id": "native-session-id" },
   "other_stuff" {
      "foo": 100,
      "bar": "bizz",
   },
   "more_stuff": "fuzz"
}

*/
```

With this approach we **won't be depending on concrete event** types and only on the minimal part of schema shared by **all events**.

Reply:
good idea, let's do this.

we don't even need a new type for this, we can manipulate Web RUM event JSON and convert it to `CodableValue` and then write to the disk.

Reply:
> we don't even need a new type for this, we can manipulate Web RUM event JSON and convert it to `CodableValue` and then write to the disk like Web Logs


Sounds good 👍